crates/metrics/metrics-derive/src/expand.rs

```diff
@@ -41,14 +44,17 @@ pub(crate) fn derive(node: &DeriveInput) -> Result<proc_macro2::TokenStream> {
                         quote! {
                             #field_name: #registrar(#metric_name),
                         },
+                        quote! {
+                            #field_name: #registrar(#metric_name, labels.clone()),
+                        },
                         quote! {
                             #describe(#metric_name, #description);
                         },
                     ))
                 })
                 .collect::<Result<Vec<_>>>()?
                 .into_iter()
-                .unzip();
+                .unzip_n_vec();
```

Comment:
I guess you could do the same with `fold(vec1, vec2, vec3)`

Reply:
ah thank you for the tip! will take a stab